# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/faseehullah404/flyrank-ml-internship/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.


## 1. My lane (or freestyle) and why

I chose the **Refresh / Content Opportunity Scoring** lane because organizations often manage thousands of content pages, making it difficult to manually decide which ones should be updated first. The starter dataset already contains content performance metrics and trend information that can help prioritize pages for review. This lane focuses on supporting content teams by ranking pages based on their refresh opportunity rather than replacing human decision-making.

In [5]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.
import pandas as pd

df = pd.read_csv("../../data/raw/content_refresh_anonymized.csv")

print("Rows, Columns:", df.shape)
print("\nColumns:\n")
print(df.columns.tolist())

Rows, Columns: (30000, 44)

Columns:

['content_id', 'client_id', 'search_volume', 'competition', 'competition_level', 'cpc', 'content_type', 'main_intent', 'word_count', 'char_count', 'provider_used', 'model_used', 'impressions_90d', 'clicks_90d', 'pageviews_90d', 'sessions_90d', 'users_90d', 'engaged_sessions_90d', 'ai_sessions_90d', 'scroll_events_90d', 'days_with_impressions', 'days_with_sessions', 'impressions_last_30d', 'clicks_last_30d', 'sessions_last_30d', 'impressions_prev_30d', 'clicks_prev_30d', 'sessions_prev_30d', 'content_age_days', 'age_tier', 'age_tier_order', 'days_since_last_update', 'freshness_tier', 'word_count_tier', 'char_count_tier', 'ctr', 'avg_position', 'engagement_rate', 'scroll_rate', 'ai_traffic_pct', 'impression_tier', 'position_tier', 'trend_direction', 'trend_pct']


## 2. The question: decision, action, cost of a wrong call

**Decision:** Which content pages should be reviewed and refreshed first?

**Who acts?** A content manager or SEO specialist.

**Action:** Review the highest-priority pages, update their content, improve relevance, or decide whether they should be expanded, monitored, or retired.

**Cost of a wrong recommendation:** If an important page is overlooked, it may continue losing traffic and engagement. If a healthy page is incorrectly prioritized, the team may waste valuable time and resources on unnecessary updates.

**Why data or ML helps:** With 30,000 content items, manually evaluating every page is impractical. A scoring system that considers multiple signals can help prioritize pages more consistently than manual inspection alone.

In [ ]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [9]:
import pandas as pd

df = pd.read_csv("../../data/raw/content_refresh_anonymized.csv")

print("Total content items:", len(df))
print("Unique clients:", df["client_id"].nunique())

declining_pct = (
    df["trend_direction"]
      .value_counts(normalize=True)["down"] * 100
)

print(f"Declining pages: {declining_pct:.2f}%")

Total content items: 30000
Unique clients: 32
Declining pages: 54.21%


## Quick look at the data

The starter dataset contains **30,000 content items** from **32 anonymized clients**. Out of these, **54.21% of the pages are marked as declining**, indicating that a large portion of the content may require attention. Since manually reviewing thousands of pages is impractical, this dataset provides a good opportunity to build a decision-support system that helps prioritize which pages should be refreshed first.

## 4. Careful words: what I can and can't claim

This project aims to provide **decision support** by identifying pages that may deserve attention based on observed patterns in the available data. Any findings will be interpreted as **observed** and **directional**, not as proof of cause and effect. The project will not claim to predict Google's ranking algorithm or guarantee that refreshing a page will improve its performance. Instead, it will help prioritize content review using evidence from historical data.

In [ ]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.